### PR TITLE
SNOW-949967 Add an optional offset token parameter for openChannel

### PR DESCRIPTION
**Pull request by sfc-gh-asen**
Adds an optional offsetToken field to OpenChannelRequest.java which will overwrite existing offset token if provided.

### CODE REVIEW

**Review by sfc-gh-tjones — APPROVED** [review]

File: src/main/java/net/snowflake/ingest/streaming/OpenChannelRequest.java [hunk 1]
```diff
@@ -40,6 +40,9 @@ public enum OnErrorOption {
   // Default timezone for TIMESTAMP_LTZ and TIMESTAMP_TZ columns
   private final ZoneId defaultTimezone;
 
+  private final String offsetToken;
+  private final boolean isOffsetTokenProvided;
```
Comment (sfc-gh-tzhang) [inline src/main/java/net/snowflake/ingest/streaming/OpenChannelRequest.java:44]: Could you explain why isOffsetTokenProvided is required?

Comment (sfc-gh-tjones) [reply src/main/java/net/snowflake/ingest/streaming/OpenChannelRequest.java:44]: I think this is to distinguish null as an offset token vs. it not being set.

Comment (sfc-gh-asen) [reply src/main/java/net/snowflake/ingest/streaming/OpenChannelRequest.java:44]: Exactly, i want to support setting the offsetToken as null as well.

 In order to distinguish between default null string vs user provided null, I am using this extra flag.

Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/OpenChannelRequest.java:44]: I see, could you add a IT test for NULL?
